pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -887,6 +887,9 @@ protected void doTakeSnapshot() {
         ServerGauge.UPSERT_VALID_DOC_ID_SNAPSHOT_COUNT, numImmutableSegments);
     _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId,
         ServerGauge.UPSERT_PRIMARY_KEYS_IN_SNAPSHOT_COUNT, numPrimaryKeysInSnapshot);
+    _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId,
```

Comment:
Can we emit it only when it is not 0? Let's also add a warning log when it happens

Reply:
Thanks for the suggestion! Updated!